disnake/ext/commands/params.py

```diff
@@ -883,7 +883,11 @@ def isolate_self(
         parametersl.pop(0)
     if parametersl:
         annot = parametersl[0].annotation
-        if issubclass_(annot, ApplicationCommandInteraction) or annot is inspect.Parameter.empty:
+        if (
+            issubclass_(annot, ApplicationCommandInteraction)
+            or annot is inspect.Parameter.empty
+            or annot.__class__.__name__ == "_GenericAlias"
```

Comment:
Instead of checking `__name__` (which is error-prone, e.g. `List[int]` would also match), this should use `get_origin` beforehand like so to get the "underlying" type of `annot`:
https://github.com/DisnakeDev/disnake/blob/b73a359ed34ddc934ae7d488f817277eeb320304/disnake/ui/button.py#L322-L323